### PR TITLE
roachprod: load balancer from pgurl command

### PR DESCRIPTION
**Pull request by herkolategan**
Previously, the load balancer IP or URL can only be retrieved using the expanders. This change adds the ability to also get the URL from the `roachprod` pgurl command by using the `lb` node selector similar to how it is done with expansion.

Epic: None
Release Note: None

### CODE REVIEW

File: pkg/roachprod/install/nodes.go [hunk 1]
```diff
@@ -44,7 +44,9 @@ func ListNodes(s string, numNodesInCluster int) (Nodes, error) {
 		return nil, errors.AssertionFailedf("invalid number of nodes %d", numNodesInCluster)
 	}
 
-	if s == "all" {
+	// "L" is a special value that also returns all nodes, but implies a load
+	// balancer should be used when applicable.
+	if s == "all" || s == "L" {
```
Comment (srosenberg) [inline pkg/roachprod/install/nodes.go:49]: Why not `LB`? `L` doesn't seem like a memorable mnemonic unlike `all`. Should it also be case-insensitive?

Comment (herkolategan) [reply pkg/roachprod/install/nodes.go:49]: I matched this to how the expanders also select the load balancer vs. node IPs. I can update both. I'll maybe add it to the command help as well to make it clear as an option.

**Review by srosenberg — APPROVED** [review]
Thanks for switching to the `lb` mnemonic; I promise it will be easier to remember (for me) :) The rest looks good, but do update the commit msg. to replace `L` with `lb`.

File: pkg/roachprod/install/nodes.go [hunk 2]
```diff
@@ -44,7 +44,9 @@ func ListNodes(s string, numNodesInCluster int) (Nodes, error) {
 		return nil, errors.AssertionFailedf("invalid number of nodes %d", numNodesInCluster)
 	}
 
-	if s == "all" {
+	// "lb" is a special value that also returns all nodes, but implies a load
+	// balancer should be used when applicable.
```
Comment (renatolabs) [inline pkg/roachprod/install/nodes.go:48]: Why "when applicable"? I don't see logic below to handle the case where a load balancer is not available (would we want that?)

File: pkg/roachprod/roachprod.go [hunk 3]
```diff
@@ -184,6 +184,13 @@ func newCluster(
 	return c, nil
 }
 
+// preferLoadBalancer determines if the node selector section in the cluster
```
Comment (renatolabs) [inline pkg/roachprod/roachprod.go:187]: I guess this is a similar point, but `prefer` suggests that actually returning a LB IP is optional, but it seems we return an error below if we can't find a LB.

I actually prefer this semantics (error returned if LB requested but doesn't exist) unless there's strong reason to provide a fallback. In that case, maybe `requestedLoadBalancer` might be a better name.

File: pkg/roachprod/install/expander.go [hunk 4]
```diff
@@ -22,8 +22,8 @@ import (
 )
 
 var parameterRe = regexp.MustCompile(`{[^{}]*}`)
-var pgURLRe = regexp.MustCompile(`{pgurl(:[-,0-9]+|:L)?(:[a-z0-9\-]+)?(:[0-9]+)?}`)
-var pgHostRe = regexp.MustCompile(`{pghost(:[-,0-9]+|:L)?(:[a-z0-9\-]+)?(:[0-9]+)?}`)
+var pgURLRe = regexp.MustCompile(`{pgurl(:[-,0-9]+|:(?i)lb)?(:[a-z0-9\-]+)?(:[0-9]+)?}`)
```
Comment (renatolabs) [inline pkg/roachprod/install/expander.go:25]: At some point it would be nice to document these with examples, these regexes are getting complicated 😄

**Review by renatolabs — APPROVED** [review]

File: pkg/roachprod/install/nodes.go (hunk 2 above)
Comment (herkolategan) [reply pkg/roachprod/install/nodes.go:48]: True, I'll update the comment to be more concise.

File: pkg/roachprod/install/expander.go (hunk 4 above)
Comment (herkolategan) [reply pkg/roachprod/install/expander.go:26]: True, these are becoming a bit over-encumbered.

**Comment by herkolategan** [comment]
TFTRs!

bors r=srosenberg,renatolabs

**Pull request merged**